### PR TITLE
Squashed commit of the following:

### PR DESCRIPTION
**Pull request by madragonse**
commit 7f809a5213ffa3bc7c61411eb7acf7a6bb7f021c
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Wed Sep 3 17:04:21 2025 +0200

    minitron code fixes

commit 84ca80a2be2609a11053b4cd1144f742215e40b8
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Fri Aug 29 14:41:27 2025 +0200

    minitron code fixes

commit 3436e5a5a12d0dc70978c8e54af129c0bc7e58d9
Merge: 06c0601 7f3210d
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Fri Aug 29 14:38:30 2025 +0200

    Merge remote-tracking branch 'origin/main' into wanda-baseline

commit 7f3210d057af1b681fd3d69b37b77dc8118100ce
Author: Maciej Stefaniak <48411094+madragonse@users.noreply.github.com>
Date:   Thu Aug 28 17:31:40 2025 +0200

    configs refactor - eval

    * Base configs+ (#20)

    * to hf +

    * FSDP2 (#25)

    fsdp2 added

    * dev

    * fixing base configs

    * nano checkpointing works

    * hf conversion fix, datt+

    * dev

    * improve logging

    * dev

    * fsdp2 fix checkpoint

    * dev

    * checkpoint path is unique

    * Delete src/core/conversion_to_hf.py

    * log environs to neptune

    * Revert commit

    * restore commit

    * revert

    * revert

    * revert

    * revert

    * fix double quote error

    * add abstract dataset

    * add abstract dataset

    * pr changes

    * cleaned older configs

    * Mas/eval config3 (#34)

    * llama compression

    * embedding fix

    * comments addressed

    * tokfix

    * llama3.1_8B

    * add support for fineweb-edu dataset

    * fix double quote error

    * add abstract dataset

    * add abstract dataset

    * dev

    * dev

    * pr update

    ---------

    Co-authored-by: Michał Krutul <crewtoolcompany@gmail.com>
    Co-authored-by: Mikołaj Dziok <m1kush@disroot.org>
    Co-authored-by: Mikołaj Dziok <78161354+m1kush@users.noreply.github.com>

    * pr changes

    ---------

    Co-authored-by: Michał <45643363+crewtool@users.noreply.github.com>
    Co-authored-by: Mikołaj Dziok <m1kush@disroot.org>
    Co-authored-by: Michał Krutul <crewtoolcompany@gmail.com>
    Co-authored-by: Mikołaj Dziok <78161354+m1kush@users.noreply.github.com>

commit 06c06012784fa2b3b493a9855ed9ebd805a7683c
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Wed Aug 27 13:15:23 2025 +0200

    add minitron pruning

commit 90f7acc7038c07e216d61e11bf83f8288cc74d12
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Wed Aug 27 13:01:34 2025 +0200

    Revert "add minitron pruning"

    This reverts commit 66b3c30681f7ae34c4cb53dde57ef419ad2b7737.

commit 66b3c30681f7ae34c4cb53dde57ef419ad2b7737
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Wed Aug 27 12:52:10 2025 +0200

    add minitron pruning

commit 93582a093b8de846939410bd55d575cc9edcd65a
Merge: 1964b2e c6b9bb9
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sun Aug 24 20:00:46 2025 +0200

    Merge remote-tracking branch 'origin/main' into wanda-baseline

commit c6b9bb923529a18c74df323e79c09bd3ba54e3c3
Merge: acea165 d3f282c
Author: Mikołaj Dziok <78161354+m1kush@users.noreply.github.com>
Date:   Sat Aug 23 14:47:20 2025 +0200

    Merge pull request #31 from llm-random/fineweb

commit d3f282ceb3f237d9a18fd9b9fd15444c371c2e28
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sat Aug 23 14:32:29 2025 +0200

    add abstract dataset

commit 436f4ef8fefb06e8223247a95187a9be1cae700a
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sat Aug 23 14:32:16 2025 +0200

    add abstract dataset

commit f06674f07bcc8a24c7129f7ed86ee6cf7eb3bd60
Merge: c9d64d1 acea165
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sat Aug 23 14:05:27 2025 +0200

    Merge remote-tracking branch 'origin/main' into fineweb

commit acea165df802a386fd85f8471022441b1d9485bc
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sat Aug 23 14:02:51 2025 +0200

    fix double quote error

commit c9d64d12340a278c32af26b659b3d985f134146c
Merge: 44b9627 0324c69
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Sat Aug 23 13:54:03 2025 +0200

    Merge remote-tracking branch 'origin/main' into fineweb

commit 0324c69875e345f7682268aa6a7f401f3b593036
Author: Maciej Stefaniak <48411094+madragonse@users.noreply.github.com>
Date:   Sat Aug 23 09:39:08 2025 +0200

    log environs to neptune

commit 38ab40485511bdfae86021e291c3af0d71c4d5a9
Author: Maciej Stefaniak <madragonse@gmail.com>
Date:   Sat Aug 23 02:03:11 2025 +0200

    checkpoint path is unique

commit 169a0881843b02b4facd5bb39bcf4221eda1749c
Author: Maciej Stefaniak <48411094+madragonse@users.noreply.github.com>
Date:   Sat Aug 23 01:33:21 2025 +0200

    fsdp2 fix checkpoint

commit 084ddba4c3ba208cef958edeef8b2292e144d744
Author: Michał <45643363+crewtool@users.noreply.github.com>
Date:   Sat Aug 23 01:15:25 2025 +0200

    improve logging

commit 44b9627fa0cacf609e767f5d0f6f2ef15c68337b
Merge: b200f37 92bbc52
Author: Mikołaj Dziok <m1kush@disroot.org>
Date:   Thu Aug 21 20:58:14 2025 +0200

    Merge remote-tracking branch 'origin/main' into fineweb

commit 92bbc52cb4c0499f68c816330b60ae8839e015e4
Author: Michał <45643363+crewtool@users.noreply.github.com>
Date:   Thu Aug 21 17:08:43 2025 +0200

    FSDP2 (#25)

    fsdp2 added

commit 25f653412b16e040941fbc344299e0a29ad67220
Author: Maciej Stefaniak <48411094+madragonse@users.noreply.github.com>
Date:   Mon Aug 18 11:40:06 2025 +0200

    Base configs+ (#20)

### CODE REVIEW

File: configs/_cluster/entropy.yaml
Comment (crewtool) [inline configs/_cluster/entropy.yaml:1]: Own PR

File: configs/_dataset/fineweb.yaml
Comment (crewtool) [inline configs/_dataset/fineweb.yaml:1]: It would be missleading

File: src/core/utils.py [hunk 1]
```diff
@@ -34,3 +34,6 @@ def cast_state_dict_to_tensors(state_dict, device="cpu"):
         else:
             full_state[k] = v
     return full_state
+
+def finish_exp(model):
```
Comment (madragonse) [inline src/core/utils.py:38]: needed for importances calc as model additional functs

File: main.py [hunk 2]
```diff
@@ -198,7 +203,10 @@ def run(cfg: OmegaConf, metric_logger=None):
         copy_llama_model_weights_from_HF(model, cfg.trainer.checkpoint.load.path)
         if cfg.get("apply_functions", None):
             for fn in instantiate(cfg.apply_functions):
-                fn(model)
+                res = fn(model)
+                if res == False:
+                    cleanup() 
+                    return 0
```
Comment (crewtool) [inline main.py:209]: Dont get it

File: configs/llama_1B_importances.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    path: ??? # change
+
+infrastructure:
+  metric_logger:
+    name: Llame_PC
+    tags:
+      - nano
+      - pc
+      - IMPORTANCS
+
```
Comment (crewtool) [inline configs/llama_1B_importances.yaml:36]: Typo

File: main.py [hunk 4]
```diff
@@ -159,7 +159,12 @@ def log_environs(metric_logger):
     for environ_key in scrap_keys:
         metric_logger.run[f"job/{environ_key}"] = str(environs.get(environ_key))
 
-def run(cfg: OmegaConf, metric_logger=None):
+def get_device():
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    return device
+
+def run(cfg, metric_logger=None):
```
Comment (crewtool) [inline main.py:167]: OmegaConf

File: configs/llama_1B_importances.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  slurm:
+    job-name: importances
+    gres: gpu:1
+    mem_per_gpu: 125G
```
Comment (crewtool) [inline configs/llama_1B_importances.yaml:40]: Not needed

**Review by crewtool — CHANGES_REQUESTED** [review]
Very nice! Fix small minor things before merge

File: configs/_dataset/fineweb.yaml
Comment (madragonse) [reply configs/_dataset/fineweb.yaml:1]: ok, moved from cluester to dataset

File: configs/llama_1B_importances.yaml (hunk 3 above)
Comment (madragonse) [reply configs/llama_1B_importances.yaml:36]: ok

File: configs/llama_1B_importances.yaml (hunk 5 above)
Comment (madragonse) [reply configs/llama_1B_importances.yaml:40]: ok

File: main.py (hunk 2 above)
Comment (madragonse) [reply main.py:209]: needed a func that can end exp

File: src/projected_compression/weight_importances.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    return dmodel_importances, dff_importances
+
+
+def _calculate_random_dimension_importances(dmodel, dff, n_blocks):
+    dmodel_importances = torch.range(0, dmodel-1)
+    dff_importances = torch.zeros(n_blocks, dff)
+    dff_importances[:,] = torch.range(0, dff-1)
+
+    return dmodel_importances, dff_importances
```
Comment (crewtool) [inline src/projected_compression/weight_importances.py:69]: Surely not random xd

File: main.py (hunk 2 above)
Comment (crewtool) [reply main.py:209]: Why?

Comment (madragonse) [reply main.py:209]: hot-fix

File: src/core/utils.py (hunk 1 above)
Comment (madragonse) [reply src/core/utils.py:38]: hotfix-as we discussed

File: src/projected_compression/weight_importances.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        handles.append(
+            block.ff_layer.layer.ff_pre_act.register_forward_hook(
+                lambda layer, inp, out, idx=block_idx: hook_ff_pre_act(layer, inp, out, idx)
+            )
+        )
+
+    # --- Run calibration data ---
+    with torch.no_grad():
+        for i, batch in enumerate(calibration_data):
+            logger.info(f"Beginning batch {i}")
```
[the remark this answers is not in the document]
Comment (madragonse) [reply src/projected_compression/weight_importances.py:54]: it is helpful info of progress, i would like to keep in, or add instead a neptune chart log, with some kind of normalized magnitude update for channels importances, but no time now

**Review by crewtool — APPROVED** [review]
lgtm
